src/librekey/rnp_key_store.cpp

```diff
@@ -338,6 +338,9 @@ rnp_key_store_merge_subkey(pgp_key_t *dst, const pgp_key_t *src, pgp_key_t *prim
     } else if (pgp_key_is_secret(src) && !pgp_key_is_locked(src)) {
         tmpkey.pkt.material = src->pkt.material;
     }
+    /* copy validity status */
+    tmpkey.valid = dst->valid && src->valid;
+    tmpkey.validated = dst->validated && src->validated && tmpkey.valid;
```

Comment:
@dewyatt It's just to make code line shorter. But, yeah, for readability it could be changed to   `&& dst->valid && src->valid`.
Or do you mean the whole logic of this assignment?

Reply:
@dewyatt This is to trigger revalidation later on. Maybe there could be better way, but, for instance, if we merge invalid key with valid, resulting merged key may both become valid (we have self-certification) or invalid (we have revocation or whatever else).

Reply:
@ni4 Ok that makes sense to me

Reply:
Oh a comment explaining that would be good though!